SPEC.md

```diff
@@ -98,61 +98,113 @@ A build tool is something which reads one or more input artifacts and writes one
+A GitBOM document describes the immediate children of an artifact in the Artifact Dependency Graph (ADG).
+
+A GitBOM document utilizes precisely one identifier type.
+
+### GitBOM identifier
+
+A GitBOM document is identified by computing its identifier as an artifact with the identifier type used for identifiers within the GitBOM document itself.
+
+#### GitBOM Document Header
+
+In order to distinguish the type of identifier used in the GitBOM document, it begins with a single newline terminated header line:
+
+```
+${identifier type uri prefix}\n
+```
+
+For example:
+
+```
+gitoid:blob:sha1\n
+```
+
+or
+
+```
+gitoid:blob:sha256\n
+```
+
+All identifiers in a GitBOM document MUST be of the identifier type declared in the header.
+
+#### GitBOM Document Child Records
+
+The GitBOM document after the header consists of a list of newline delimited child records
 
 A child artifact which is itself a leaf artifacts would be represented by
 
 ```
-blob⎵${git object id of child}\n
+blob⎵${artifact id of child}\n
 ```
 
 A child artifact which is itself a derived artifact would be represented by
 ```
-blob⎵${git object id of child}⎵bom⎵${git object id of child's GitBOM document}\n
+blob⎵${artifact id of child}⎵bom⎵${GitBOM identifier of child's GitBOM document}\n
 ```
 
-### GitBOM identifier
+The child artifact redords must be written to the GitBOM Document in lexical order.
 
-For a given artifact, the GitBOM ID of that artifact is simply the git object id of it's GitBOM Document.
+The artifact id and GitBOM document id must both be of the identifier type declared in the GitBOM document header.
 
 ### GitBOM Identifier Embedding
 
-Each build tool should embed the GitBOM ID of its output artifact(s) into the output artifact in a manner:
+Each build tool should embed into the output artifact a new line delimited, lexically ordered, list of GitBOM identifiers for each mandatory identifier type in a manner:
 
-1.  Appropriate to the type of artifact
-2.  Generally agreed upon for that artifact
+1. Appropriate to the type of artifact
+2. Generally agreed upon for that artifact
 
 ### GitBOM Document Construction by a Build Tool
 
-When a build tool reads an artifact, it should for each input file:
-1. Compute the git object id of the input file - ${artifact identifier}
-2. Examine the artifact for an embedded GitBOM identifier - ${gitbom idenfitier}
-3. If the artifact contains a GitBOM identifer it construct a record:
-   ```
-   blob⎵${artifact identifier}⎵bom⎵${GitBOM identifier}\n
-   ```
-4. Otherwise construct a record:
-   ```
-   blob⎵${artifact identifier}\n
-   ```
-
-Concatenate all records together and apply lexical ordering by line to create the GitBOM document of the
-output artifact.
+A build tool creating an output artifact must compute a GitBOM document of each mandatory artifact id type.
```

Comment:
```suggestion
A build tool creating an output artifact must compute a GitBOM Document of each mandatory artifact id type.
```

Reply:
Must or should?

Reply:
My instinct is MUST.  However, there is the corner case of 'is the build tool even doing GitBOM' to think about.